utils/src/main/kotlin/Extensions.kt

```diff
@@ -53,7 +53,11 @@ fun ByteArray.toHexString(): String = joinToString("") { String.format("%02x", i
  * Return the absolute file with a leading "~" in a Unix path expanded to the current user's home directory.
  */
 fun File.expandTilde(): File =
-    File(path.replace(Regex("^~/"), "${System.getProperty("user.home")}/")).absoluteFile
+    if (System.getenv("SHELL") != null) {
+        File(path.replace(Regex("^~"), Regex.escapeReplacement(System.getProperty("user.home"))))
+    } else {
+        this
+    }.absoluteFile
```

Comment:
You should adapt the docs, if this now handles "\\" it's not only for Unix paths anymore.
Could you have used `File.separator` instead in the pattern to make it work?

Reply:
It's a bit more complicated: I didn't want the replacement to happen in `cmd` on Windows (because there `~` could be a regular character in a file / directory name), so I couldn't just use `File.separator` (without any further checks). Also, from a user perspective this still only handles Unix-style paths, because in the user's shell `/` is used as a path separator, even if the `File` class always uses `\` on Windows.